### DISCUSSION TITLE
Question: Are location points buffered while off-line?

### DISCUSSION THREAD

**Discussion by muety** [owntracks/android#1106] [Q&A] [answered]
I am sure the app is capable of this, but I couldn't find any details in the documentation. Does the app "cache" data points while my phone is offline and then flush them when network is available again? If yes, how many points will be cached at max. and (how) can I changed that parameter?

**Comment by growse** [comment, marked as answer]
Yes.

If the app fails to send a location message, it's retried (with an exponential backoff capped at every 2 mins, I think). It will retry to send each message a fixed number of times, before dropping it and moving onto the next message. For location messages, it's actually retried effectively for a week (about 10,000 times), because there are some users who are out of network contact for multiple days.

You should be able to see the size of the current location queue in the status page. The maximum size of the queue is currently 10,000 points.

Neither of these two values are configurable - I'd be interested in understanding the reasons behind a desire to make them configurable?

**Reply by woj-tek** [reply]
(Apologies for digging up the old topic but was just searching for information about cache and it came up)

> If the app fails to send a location message, it's retried (with an exponential backoff capped at every 2 mins, I think). It will retry to send each message a fixed number of times, before dropping it and moving onto the next message. For location messages, it's actually retried effectively for a week (about 10,000 times), because there are some users who are out of network contact for multiple days.

Is the exponential backoff capped in any way?
Will the data be submitted at once when the device connects?



> Neither of these two values are configurable - I'd be interested in understanding the reasons behind a desire to make them configurable?

Would it be possible to have configurable in the context of days and not the data points?

As to why: for some "about a week" may not be enough. And then again - the datapoints data itself shouldn't take up that much space.
